internal/integration/infra_sdk_emitter.go

```diff
@@ -0,0 +1,181 @@
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+	for _, me := range metrics {
+		switch me.metricType {
+		case metricType_GAUGE:
+			err = e.emitGauge(i, me, now)
+			break
+		case metricType_COUNTER:
+			err = e.emitCounter(i, me, now)
+			break
+		case metricType_SUMMARY:
+			err = e.emitSummary(i, me, now)
+			break
+		case metricType_HISTOGRAM:
+			err = e.emitHistogram(i, me, now)
+			break
+		default:
+			err = fmt.Errorf("unknown metric type %q", me.metricType)
+		}
+
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to create metric from '%s'", me.name)
+		}
+	}
+
+	err = i.Publish()
+	return err
+}
+
+func (e *InfraSdkEmitter) emitGauge(i *integration.Integration, metric Metric, timestamp time.Time) error {
+	m, err := integration.Gauge(timestamp, metric.name, metric.value.(float64))
+	if err != nil {
+		return err
+	}
+	addDimensions(m, metric.attributes)
+	i.HostEntity.AddMetric(m)
+
+	return nil
+}
+
+func (e *InfraSdkEmitter) emitCounter(i *integration.Integration, metric Metric, timestamp time.Time) error {
+	m, err := integration.Count(timestamp, metric.name, metric.value.(float64))
+	if err != nil {
+		return err
+	}
+	addDimensions(m, metric.attributes)
+	i.HostEntity.AddMetric(m)
+
+	return nil
+}
+
+func (e *InfraSdkEmitter) emitHistogram(i *integration.Integration, metric Metric, timestamp time.Time) error {
+	hist, ok := metric.value.(*dto.Histogram)
+	if !ok {
+		return fmt.Errorf("unknown histogram metric type for %q: %T", metric.name, metric.value)
+	}
+
+	if sumDelta, ok := e.deltaCalculator.CountMetric(metric.name+"_sum", metric.attributes, hist.GetSampleSum(), timestamp); ok {
+		m, err := integration.Summary(timestamp, metric.name+"_sum",
+			1,
+			math.NaN(), // can we calc average here?
+			sumDelta.Value,
+			math.NaN(),
+			math.NaN(),
+		)
+		if err != nil {
+			return err
+		}
+		addDimensions(m, metric.attributes)
+		i.HostEntity.AddMetric(m)
+	}
+
+	m, err := integration.Count(timestamp, metric.name+"_count", float64(hist.GetSampleCount()))
+	if err != nil {
+		return err
+	}
+	addDimensions(m, metric.attributes)
+	i.HostEntity.AddMetric(m)
+
+	metricName := metric.name + "_bucket"
+	for _, b := range hist.GetBucket() {
+		m, err = integration.Count(timestamp, metricName, float64(b.GetCumulativeCount()))
+		if err != nil {
+			return err
+		}
+		addDimensions(m, metric.attributes)
+		_ = m.AddDimension("le", fmt.Sprintf("%g", b.GetUpperBound()))
+
+		i.HostEntity.AddMetric(m)
+	}
+	return nil
+}
+
+func (e *InfraSdkEmitter) emitSummary(i *integration.Integration, metric Metric, timestamp time.Time) (err error) {
+	summary, ok := metric.value.(*dto.Summary)
+	if !ok {
+		return fmt.Errorf("unknown summary metric type for %q: %T", metric.name, metric.value)
+	}
+
+	m, err := integration.Summary(timestamp, metric.name+"_sum",
+		1,
+		math.NaN(),             // can we calc average here?
+		summary.GetSampleSum(), // not sure the delta gets calculated in the agent for summary metrics
```

Comment:
should we verify this comment before merging?

Reply:
I think we can remove them. It was copied basically from the other emitter so regarding the average I think we could, but the spec for the NR prometheus "consumer" did not include it: https://source.datanerd.us/core-data-platform/prometheus-consumer#conversion-rules
Regarding the second comment, I'm still not sure if it will work if we do nothing. I don't think the agent will do anything, it will only calculate deltas for counters. But the telemetry emitter also does not calculate the delta so let's keep it as is and when we get everything working check if the values are correct